src/Core/Tool/Authorizer/ContactAuthorizer.php

```diff
@@ -31,12 +32,20 @@ class ContactAuthorizer implements Authorizer
 	// It uses `PrivAccess` to provide the `getAllowedPrivs` method.
 	use PrivAccess;
 
+	// It uses `PrivateDeployment` to check whether a deployment is private
+	use PrivateDeployment;
+
 	/* Authorizer */
 	public function isAllowed(Entity $entity, $privilege)
 	{
 		// These checks are run within the user context.
 		$user = $this->getUser();
 
+		// Only logged in users have access if the deployment is private
+		if ($this->isPrivate() and !$user->getId()) {
```

Comment:
In fact, given you're doing this check everywhere, maybe alias it into the PrivateDeployment trait too. ie.
Add `canAccessDeployment` (or a better name) and do
```
if (! $this->canAccessDeployment()) {
  return false;
}
```
or maybe thats overkill?